### PR TITLE
[ pwm, top_earlgrey ] Adding PWM to Earlgrey

### PR DESCRIPTION
**Pull request by martin-lueker**
Merging one instance of pwm into aon domain for top_earlgey.

### CODE REVIEW

**Comment by martin-lueker** [comment]
Hi @rswarbrick, 
  Saw your note in Slack somewhere.  This can be registered as "not WIP".  There is still a VCS/Xcelium difference that we are trying to sort out for CI. However that is currently settling behind the push to get SPI_HOST into a PR.
 
The PR is ready for review, but most of it is repeated in the parent branch in #5556.  Once that is merged this PR will be a lo easier to review.

File: hw/top_earlgrey/data/top_earlgrey.hjson [hunk 1]
```diff
@@ -440,6 +440,14 @@
...
       domain: "Aon",
       base_addr: "0x40440000"
+    },  
+    { name: "pwm_aon",
+      type: "pwm",
+      clock_srcs: {clk_i: "io_div4", clk_core_i: "aon"},
+      clock_group: "powerup",
+      reset_connections: {rst_ni: "sys_io_div4", rst_core_ni: "sys_aon"},
+      domain: "Aon",
+      base_addr: "0x40450000",
```
Comment (msfschaffner) [inline hw/top_earlgrey/data/top_earlgrey.hjson:450]: I think you are missing the associated entry in the crossbar configuration (xbar_peri.hjson).

Comment (martin-lueker) [reply hw/top_earlgrey/data/top_earlgrey.hjson:450]: Oh, my! Thanks for the feedback.   I think you are right.  I forgot that this time.  That may explain why the CI is failing.

**Comment by martin-lueker** [comment]
Status update:  I'm trying to get spi_host in today.  Once that's done I'll loop around and fix @rswarbrick's suggestions.

**Comment by martin-lueker** [comment]
This has been rebased to remove the PWM implementation files.  This now just contains:
- The top_earlgrey.hjson and peri.hjson changes
- Relevant autogens
- A patch to fix CI errors caused by #6059

**Comment by martin-lueker** [comment]
So many thanks to @tjaychen for pointing out that (at least as of right now?) there are no CI errors with this one.    Can someone take a look at the content?

**Review by tjaychen — APPROVED** [review]

**Review by msfschaffner — APPROVED** [review]

**Pull request merged**